src/main/java/org/testng/internal/Parameters.java

```diff
@@ -43,6 +38,42 @@
  */
 public class Parameters {
   public static final String NULL_VALUE= "null";
+  private static Class<?>[] annotationList = new Class<?>[] {
+      BeforeSuite.class,
+      AfterSuite.class,
+      BeforeTest.class,
+      AfterTest.class,
+      BeforeClass.class,
+      AfterClass.class,
+      BeforeGroups.class,
+      AfterGroups.class,
+      BeforeMethod.class,
+      AfterMethod.class
+  };
+
+  private static Map<String, List<Class>> mapping = new HashMap<>();
```

Comment:
I have changed this to 

`private static Map<String, List<Class>> mapping = Maps.newHashMap();`

The other option will not work because the keys are annotations and Not the values in the map.